### PR TITLE
auth: ellipsis + right click options for `Developer Tools`

### PR DESCRIPTION
**Pull request by nkomonen-amazon**
This commit allows user to right click to root CW or CC
Node or click the new ellipsis icon, and they will get
the options to Add Connection or Signout (if appropriate)

Screenshot 2023-10-20 at 10 50 42 AM

Kapture 2023-10-20 at 10 36 27

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/codecatalyst/auth.ts [hunk 1]
```diff
@@ -45,6 +45,10 @@ export const isValidCodeCatalystConnection = (conn: Connection): conn is SsoConn
 export const isUpgradeableConnection = (conn: Connection): conn is SsoConnection =>
     isBuilderIdConnection(conn) && !isValidCodeCatalystConnection(conn)
 
+export function setCodeCatalystConnectedContext(isConnected: boolean) {
+    return vscode.commands.executeCommand('setContext', 'CODECATALYST_CONNECTED', isConnected)
```
Comment (justinmk3) [inline src/codecatalyst/auth.ts:49]:
```suggestion
    return vscode.commands.executeCommand('setContext', 'aws.codecatalyst.connected', isConnected)
```

File: package.json [hunk 2]
```diff
@@ -1981,6 +1998,32 @@
...
                 }
+            ],
+            "aws.codecatalyst.submenu": [
+                {
+                    "command": "aws.codecatalyst.manageConnections",
+                    "group": "0@1"
+                },
+                {
+                    "command": "aws.codecatalyst.signout",
+                    "when": "!isCloud9 && CODECATALYST_CONNECTED",
```
Comment (justinmk3) [inline package.json:2009]: setContext doesn't actually work in cloud9, but it looks like all of these cases are `!isCloud9` so it doesn't matter in this case.

File: src/codecatalyst/auth.ts [hunk 3]
```diff
@@ -101,10 +112,6 @@ export class CodeCatalystAuthenticationProvider {
         }
     }
 
-    public async removeSavedConnection() {
-        await this.secondaryAuth.removeConnection()
```
Comment (justinmk3) [inline src/codecatalyst/auth.ts:105]: thank you for lifting these special cases into a higher layer!

**Review by justinmk3 — APPROVED** [review]

**Review by andrewyuq — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (nkomonen-amazon) [reply package.json:2009]: Better to have less uses of `isCloud9`, will remove. Good callout

**Pull request merged**